python-backend/app/api/parties/party_appt/resources/mine_party_appt_resource.py

```diff
@@ -74,6 +74,13 @@ def post(self, mine_party_appt_guid=None):
             new_mpa.save()
         except AssertionError as e:
             self.raise_error(400, 'Error: {}'.format(e))
+        except alch_exceptions.IntegrityError as e:
+            if "daterange_excl" in str(e):
+                mpa_type_name = MinePartyAppointmentType.find_by_mine_party_appt_type_code(
+                    data.get('mine_party_appt_type_code')).description
+                self.raise_error(500, f'Error: Date ranges for {mpa_type_name} must not overlap')
```

Comment:
Can this be more instructional, like maybe "Start and end dates for this {mpa_type_name} overlap with an existing {mpa_type_name}. Please update the existing {mpa_type_name}'s stand and end dates before adding a new {mpa_type_name} for this period. ?" Check with Aaron.